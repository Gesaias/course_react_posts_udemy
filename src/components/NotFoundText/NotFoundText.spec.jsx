import { render, screen } from '@testing-library/react';
import { NotFoundText } from '.';

describe('<NotFoundText />', () => {
  it('should render NotFoundText correctly non-centered', () => {
    render(<NotFoundText center={false} />);

    expect.assertions(1);

    expect(screen.getByText(/Nenhum post encontrado!!/i)).toBeInTheDocument();
  });

  it('should render NotFoundText correctly centered', () => {
    render(<NotFoundText center={true} />);

    expect.assertions(1);

    expect(screen.getByText(/Nenhum post encontrado!!/i)).toHaveAttribute('class', 'center');
  });

  // Snapshot

  it('should match snapshot', () => {
    const container = render(<NotFoundText center={true} />).container;
    const { firstChild } = container;

    expect(firstChild).toMatchSnapshot();
  });
});
