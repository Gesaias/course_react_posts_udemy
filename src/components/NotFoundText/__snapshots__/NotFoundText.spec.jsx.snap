// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFoundText /> should match snapshot 1`] = `
<p
  class="center"
>
  Nenhum post encontrado!!
</p>
`;
